### ISSUE TITLE
Help：repeated insertion of the frontmatter containing a banner field

### ISSUE THREAD

**Issue by wuzhen97** [noatpad/obsidian-banners#86]
Every time I open a note, the plug-in will automatically add a frontmatter containing a banner, but this note originally has a frontmatter containing a banner and other fields.

**I tried in some public sample libraries and found that the notes in the sample vault（like [BT examples](https://github.com/cumany/Blue-topaz-examples)） did not have this problem. But when I copy my notes to their library and then open it here, an error will occur：**

please help me，thanks~

# Some information：

<img src="https://user-images.githubusercontent.com/61906420/211011688-904b7523-3676-4bdb-ba26-7a93f3132560.png" width="400px">

## original frontmatter

```
---
citeKey: _-_2021-1
entry: "[zotero](zotero://select/items/@_-_2021-1)"
year: 2021
DOI: 10.27162/d.cnki.gjlin.2021.000208
banner: "![[013.jpg]]"
create: 2022-08-29
update: 2022-08-29
status: 
publish: false
priority: 13
aliases:
  - aliases1
tags:
  - tags1
---
```

# changed frontmatter

```
---
banner: "![[013.jpg]]"
---
---
citeKey: _-_2021-1
entry: "[zotero](zotero://select/items/@_-_2021-1)"
year: 2021
DOI: 10.27162/d.cnki.gjlin.2021.000208
banner: "![[013.jpg]]"
create: 2022-08-29
update: 2022-08-29
status: 
publish: false
priority: 13
aliases:
  - aliases1
tags:
  - tags1
---
```

**Comment by wuzhen97** [comment]
When I close all plug-ins except banners, this problem will still occur, but only for those notes that have not had the above problem before.

**Comment by wuzhen97** [comment]
OS：Windows10
Obsidian：v1.1.9
Plugin：latest

**Comment by oliviercarizzoni** [comment]
Same here, @noatpad are you still maintaining this plugin ? This error is really messing up plugin usage... Let us know

**Comment by ScottKirvan** [comment]
Same problem here - tried everything I can think of at this point, but the additional YAML keeps getting inserted above the yaml that's already in the page.

**Comment by ku-zhen** [comment]
hey, guys. 
I've avoided this problem for now by commenting out the function for inserting code. At the same time, I'm not free to choose the banner image, but fortunately I can use the templater plugin to randomly generate a banner when generating notes

**Comment by oliviercarizzoni** [comment]
> hey, guys. 
> I've avoided this problem for now by commenting out the function for inserting code. At the same time, I'm not free to choose the banner image, but fortunately I can use the templater plugin to randomly generate a banner when generating notes

Thanks for the feedback, but what is exactly that insertion function ?

**Comment by oliviercarizzoni** [comment]
Great thank you !

**Comment by zjgzimmerman** [comment]
I'm wondering if this is due to the newline characters at the end of the lines in the YAML block. 
I added banners to some notes on a linux machine then noticed that this bug would only occur when opening the notes on a windows machine. This may also explain why some users were suspecting the git plugin was causing the bug if they committed the changes from linux then opened the vault in windows.

I don't know typescript, but I checked the upsertBannerData function and saw it was splitting lines of the file on '\n'. Does that need to be \r\n for windows? I saw a thread on SO where it was recommended to split with `.split(/\r?\n/);` in order to handle windows and linux

**Comment by zjgzimmerman** [comment]
I tested changing the newline split and it looks like that fixed the issue for me. Created pull request #97

**Comment by ku-zhen** [comment]
@zjgzimmerman  oh，that's great，could you share a release version? The original author doesn't seem to maintain it anymore. thanks~

**Comment by oliviercarizzoni** [comment]
> I tested changing the newline split and it looks like that fixed the issue for me. Created pull request #97

Well done.

**Comment by zjgzimmerman** [comment]
I just added a new release [here](https://github.com/zjgzimmerman/obsidian-banners/releases)

**Comment by ku-zhen** [comment]
> I just added a new release [here](https://github.com/zjgzimmerman/obsidian-banners/releases)

thanks for your reply，but the problem is not solved for me. There may be other factors.

![image](https://user-images.githubusercontent.com/61906420/222022639-72eb5b65-40c2-4726-9917-778eac89c51c.png)

**Comment by ku-zhen** [comment]
The version you shared may have fixed a bug in some way. But for me  it's may because of the yaml format, because I found that if the banner format does not strictly conform to his convention, he will modify/insert it, and sometimes it insert it outside the yaml！

![image](https://user-images.githubusercontent.com/61906420/222023984-c7346171-5c76-411c-ac8e-d9d61aad1d28.png)

![image](https://user-images.githubusercontent.com/61906420/222023858-c9eb73e0-09f2-4dd8-86e0-1a5446aac140.png)

**Comment by ku-zhen** [comment]
Therefore, if you use plug-ins like **projects or db folder** that modify the content of your yaml, it's likely that the two formats will conflict and cause bugs

**Comment by zjgzimmerman** [comment]
Yeah, that looks like a different issue than what I was experiencing but I'm thinking your issue is also related to how it's parsing the YAML in upsertBannerData.

**Comment by zjgzimmerman** [comment]
Looks like it intentionally adds the quotes if they're not there already, found these comments around where that happens

`// Helper to wrap banner source in quotes if not already (Patch for previous versions)`
`// Wrap banner source in quotes to prevent errors later in CM6 extension`

For the spaces it looks like it's splitting on ": " and doesn't trim the second space so the path doesn't match the path passed in to the function which causes it to update the field.

Not sure what's causing the issue in the first screenshot
